pkg/build/generator/generator.go

```diff
@@ -432,6 +432,12 @@ func (g *BuildGenerator) generateBuildFromConfig(ctx kapi.Context, bc *buildapi.
 	}
 
 	build.Name = getNextBuildName(bc)
+	if _, err := g.Client.GetBuild(ctx, build.Name); err == nil {
```

Comment:
I think the resulting conflict is intentional to make sure that we don't race and create two builds accidentally
@csrwng